crates/bevy_animation/src/lib.rs

```diff
@@ -295,8 +295,18 @@ fn entity_from_path(
     // PERF: finding the target entity can be optimised
     let mut current_entity = root;
     path_cache.resize(path.parts.len(), None);
-    // Ignore the first name, it is the root node which we already have
-    for (idx, part) in path.parts.iter().enumerate().skip(1) {
+
+    let mut parts = path.parts.iter().enumerate();
+
+    // check the first name is the root node which we already have
+    let Some((_, root_name)) = parts.next() else {
+        return None;
+    };
+    if names.get(current_entity) != Ok(root_name) {
+        return None;
+    }
```

Comment:
```suggestion
    if names.get(current_entity).ok() != parts.next().map(|(_, part)| part) {
        return None;
    };
```
Just a style nit, but I would prefer this

Reply:
it's slightly different semantically - this would pass if parts is empty and the node has no name. neither of those things can happen via the gltf loader but could happen for clips/players constructed another way.

could maybe go with `if names.get(current_entity) != Ok(parts.next()?.1) {` ... i'm not sure it's nicer.